utils/DeployUtils.ps1

```diff
@@ -161,63 +161,141 @@ function ConfigureScubaGearModule{
-    Get-ChildItem $ModulePath -Include *.psd1,*psm1,*.ps1 -Recurse |
-    Set-AuthenticodeSignature -Certificate $Cert -TimeStampServer $TimeStampServer -HashAlgorithm $HashAlgorithm
+    $SignArguments = @(
+        'sign',
+        '-coe',
+        '-fd',"sha256",
+        '-tr', $TimeStampServer,
+        '-kvu',$AzureKeyVaultUrl,
+        '-kvc',$CertificateName,
+        '-kvm'
+        '-ifl',$FileList         
+    )
 
-    New-FileCatalog -Path $ModulePath -CatalogFilePath $CatalogPath -CatalogVersion 2.0 -Verbose
-    Set-AuthenticodeSignature -FilePath $CatalogPath -Certificate $Cert -TimestampServer $TimeStampServer -HashAlgorithm $HashAlgorithm
+    Write-Debug "Calling AzureSignTool: $SignArguments"
 
-    $TestResult = Test-FileCatalog -CatalogFilePath $CatalogPath
-    return 'Valid' -eq $TestResult.Status
+    $ToolPath = (Get-Command AzureSignTool).Path  
+    powershell -Command "& $ToolPath $SignArguments"      
 }
-
-function New-CodeSigningCertificate{
+function SignScubaGearModule{
     <#
-        .NOTES
-        Internal helper function
+    .SYNOPSIS
+    Code sign the specified module
+    .Description
+    This function individually signs PowerShell artifacts (i.e., *.ps1, *.pms1) and creates a 
+    signed catalog of the entire module using a certificate housed in an Azure key vault.
+    .Parameter AzureKeyVaultUrl
+    The URL of the key vault with the code signing certificate
+    .Parameter CertificateName
+    The name of the code signing certificate
+    .Parameter ModulePath
+    The root path of the module to be signed
+    .Parameter TimeStampServer
+    Time server to use to timestamp the artifacts
     #>
-    New-SelfSignedCertificate -DnsName cisa.gov -Type CodeSigning -CertStoreLocation Cert:\CurrentUser\My
-}
+    param (
+        [Parameter(Mandatory=$true)]
+        [ValidateScript({[uri]::IsWellFormedUriString($_, 'Absolute') -and ([uri] $_).Scheme -in 'https'})]
+        [System.Uri]
+        $AzureKeyVaultUrl,
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $CertificateName,
+        [Parameter(Mandatory=$true)]
+        [ValidateScript({Test-Path -Path $_ -PathType Container})]
+        $ModulePath,
+        [Parameter(Mandatory=$false)]
+        [ValidateScript({[uri]::IsWellFormedUriString($_, 'Absolute') -and ([uri] $_).Scheme -in 'http','https'})]
+        $TimeStampServer = 'http://timestamp.digicert.com'
+    )
+    
 
-function GetCodeSigningCertificate{
-    <#
-        .NOTES
-        Internal helper function
-    #>
-    #TODO:  Replace with official signing certificate
-    Get-ChildItem Cert:\CurrentUser\My -CodeSigningCert | Where-Object { $_.HasPrivateKey -and ($_.NotAfter -gt (Get-Date))}
+    # Digitally sign scripts, manifest, and modules
+    $FileList = CreateFileList -SourcePath $ModulePath -Extensions "*.ps1","*.psm1","*.psd1"
+    CallAzureSignTool @PSBoundParameters -FileList $FileList
```

Comment:
Do we need to sign both the catalog and the files?  Its been a bit but I thought if we signed the catalog we at least didn't need to sign the module files (we may need to sign the scripts)?

In any event, can you please include a link to some documentation on the requirements for which files require signing.

Reply:
I did struggle with this - there appears to be limited or at least difficult to find documentation on how to properly sign a PowerShell module to be published to PSGallery.   
[This](https://learn.microsoft.com/en-us/powershell/gallery/concepts/publishing-guidelines?view=powershellget-3.x) show general guidance that supports your understanding.

Anecdotal: I did look at Microsoft.PowerApps.PowerShell (v1.0.34) and see both *.psd1 and *.psm1 files are signed and a catalog provided. 

There are a number of Non-authoritive references such as below showing all ps1, psm1, and psd1 being signed first then cataloged. 
[Snippet](https://github.com/dell/OpenManage-PowerShell-Modules/blob/main/Sign-Module.ps1)

Reply:
Ok.  Please include a comment and a link to these references to support the decision to sign everything, just so this is not lost to the sands of time.

Reply:
Added NOTE to function documentation to cover rationale.